app/models/setting.rb

```diff
@@ -3,107 +3,66 @@ class Setting < ApplicationRecord
   validates :code, presence: true, uniqueness: true
   validates :description, presence: true
   validates :value, presence: true
+  validates :value_format, presence: true
+  validate :valid_value_format
 
-  def self.auction_currency
-    Setting.find_by(code: :auction_currency).value
-  end
+  VALUE_FORMATS = {
+    string: :string_format,
+    integer: :integer_format,
+    boolean: :boolean_format,
+    hash: :hash_format,
+    array: :array_format,
+  }.with_indifferent_access.freeze
 
-  def self.auction_minimum_offer
-    Setting.find_by(code: :auction_minimum_offer).value.to_i
+  def retrieve
+    method = VALUE_FORMATS[value_format]
+    send(method)
   end
 
-  def self.terms_and_conditions_link
-    value = Setting.find_by(code: :terms_and_conditions_link).value
-    JSON.parse(value)[I18n.locale.to_s]
+  def valid_value_format
+    formats = VALUE_FORMATS.with_indifferent_access
+    errors.add(:value_format, :invalid) unless formats.keys.any? value_format
   end
 
-  def self.default_country
-    Setting.find_by(code: :default_country).value
+  def string_format
+    return auction_duration_value if code == 'auction_duration'
+    return auctions_start_at_value if code == 'auctions_start_at'
+
+    value
   end
 
-  def self.payment_term
-    Setting.find_by(code: :payment_term).value.to_i
+  def integer_format
+    value.to_i
   end
 
-  def self.registration_term
-    Setting.find_by(code: :registration_term).value.to_i
+  def boolean_format
+    value == 'true'
   end
 
-  def self.auction_duration
-    value = Setting.find_by(code: :auction_duration).value
+  def hash_format
+    return JSON.parse(value)[I18n.locale.to_s] if code == 'terms_and_conditions_link'
+    return JSON.parse(value)[I18n.locale.to_s] if code == 'violations_count_regulations_link'
```

Comment:
How about to make a separate method for this?
Something like `localised_value`? We are returning the same value of hash in two different conditions - maybe we could join them somehow?

Reply:
Might be even better. Thanks for suggestion.